qiskit_nature/problems/second_quantization/lattice/models/heisenberg_model.py

```diff
@@ -0,0 +1,131 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Heisenberg model."""
+
+import numpy as np
+import logging
+from typing import Optional, Tuple
+from fractions import Fraction
+from qiskit_nature.second_q.operators import SpinOp
+from qiskit_nature.problems.second_quantization.lattice.lattices import Lattice
+from .lattice_model import LatticeModel
+
+logger = logging.getLogger(__name__)
+
+
+class HeisenbergModel(LatticeModel):
+    """The Heisenberg model."""
+
+    def __init__(
+        self,
+        lattice: Lattice,
+        coupling_constants: Tuple = (1.0, 1.0, 1.0),
+        ext_magnetic_field: Tuple = (0.0, 0.0, 0.0),
+    ) -> None:
+        """
+        Args:
+            lattice: Lattice on which the model is defined.
+            coupling_constants (Tuple, optional): Coupling constants. Defaults to (1.0, 1.0, 1.0).
+            ext_magnetic_field (Tuple, optional): External magnetic field. Defaults to (0.0, 0.0, 0.0).
+        """
+        super().__init__(lattice)
+        self.coupling_constants = coupling_constants
+        self.ext_magnetic_field = ext_magnetic_field
+
+    def coupling_matrix(self) -> np.ndarray:
```

Comment:
Do we need to expose the `interaction_matrix` under this new name?

Reply:
I think we don't need this, I'll remove it.